### DISCUSSION TITLE
windows: mogrify recursive batch processing

### DISCUSSION THREAD

**Discussion by aakkeemm** [ImageMagick/ImageMagick#7827] [Help]
### ImageMagick version

7.1.1-41 Q16-HDRI x64

### Operating system, version and so on

Windows 10 HOME

### Description

i am trying to run a magick mogrify batch conversion recursively in PowerShell, aiming to have all files converted to jpg in a folder hirarchy same as the originals in C:\path\sourcefolder and subfolders

`dir -recurse -include *.* | %{magick mogrify -format jpg ..\targetfolder\$_}`

doesn't obey "..\targetfolder\" but puts the jpgs into the source folders next to the original files

`dir -recurse -include *.* | %{magick mogrify -format jpg -path ..\targetfolder\ $_}`

puts all jpgs into C:\path\targetfolder discarding subfolders

Any suggestions how to maintain the folder hirarchy in targetfolder?

Alternatively the original files could be deleted in sourcefolder after conversion.

**Comment by snibgo** [comment]
I don't know Powershell, and can't advise on that.

"magick mogrify" will read files in the specified directory, and write to the directory given in "-path". It doesn't recurse into subdirectories. The "-path" directory must already exist. If it doesn't exist, then the command will fail.

**Comment by dceeee** [comment]
Something like this:
$Source = "D:\from\"
$Dest = "D:\to\"

$ErrorActionPreference = 'SilentlyContinue'
xcopy $Source $Dest /T /E
$Folders = (Get-ChildItem -Path $Source -Recurse -Directory).Fullname
$FoldersOut = (Get-ChildItem -Path $Dest -Recurse -Directory).Fullname
$Folders += $Source
$FoldersOut += $Dest
for ($i=0; $i -lt $Folders.length; $i++){
  magick mogrify -path ($FoldersOut[$i] + "\") -format jpg ($Folders[$i] + "\*.*")
  Write-Host ($Folders.length - $i)
}

**Reply by dceeee** [reply]
oops, mangled a bit, try this file

[script.zip](https://github.com/user-attachments/files/18059934/script.zip)
